src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -311,16 +311,23 @@ private void InitializeOutputFile(TOptions analyzeOptions, TContext context, Has
             {
                 InvokeCatchingRelevantIOExceptions
                 (
-                    () => aggregatingLogger.Loggers.Add(
-                            new SarifLogger(
-                                analyzeOptions.OutputFilePath,
-                                targets,
-                                analyzeOptions.Verbose,
-                                analyzeOptions.LogEnvironment,
-                                analyzeOptions.ComputeTargetsHash,
-                                Prerelease,
-                                invocationTokensToRedact: GenerateSensitiveTokensList(),
-                                invocationPropertiesToLog: analyzeOptions.InvocationPropertiesToLog)),
+                    () =>
+                    {
+                        LoggingOptions loggingOptions = LoggingOptions.None;
+
+                        if (analyzeOptions.Verbose) { loggingOptions |= LoggingOptions.Verbose; }
+                        if (analyzeOptions.LogEnvironment) { loggingOptions |= LoggingOptions.Verbose; }
+                        if (analyzeOptions.ComputeFileHashes) { loggingOptions |= LoggingOptions.Verbose; }
+
```

Comment:
I think a helper
```
private LoggingOptions AnalyzeOptionsToLoggingOptions(AnalyzeOptions analyzeOptions)
```
would be nice.

Reply:
done and authored unit tests that discovered a bug in this code. see repetition of Verbose in these conditionals. :P

---
In reply to: [135840279](https://github.com/Microsoft/sarif-sdk/pull/609#discussion_r135840279) [](ancestors = 135840279)